mlir/lib/Target/SPIRV/Deserialization/Deserializer.cpp

```diff
@@ -637,14 +637,24 @@ spirv::Deserializer::processGlobalVariable(ArrayRef<uint32_t> operands) {
 
   // Initializer.
   FlatSymbolRefAttr initializer = nullptr;
+
   if (wordIndex < operands.size()) {
-    auto initializerOp = getGlobalVariable(operands[wordIndex]);
-    if (!initializerOp) {
+    Operation *op = nullptr;
+
+    if ((op = getGlobalVariable(operands[wordIndex])))
+      initializer = SymbolRefAttr::get(
+          (dyn_cast<spirv::GlobalVariableOp>(op)).getOperation());
+    else if ((op = getSpecConstant(operands[wordIndex])))
+      initializer = SymbolRefAttr::get(
+          (dyn_cast<spirv::SpecConstantOp>(op)).getOperation());
+    else if ((op = getSpecConstantComposite(operands[wordIndex])))
+      initializer = SymbolRefAttr::get(
+          (dyn_cast<spirv::SpecConstantCompositeOp>(op)).getOperation());
```

Comment:
Since this is accessing cast results unconditionally, I'd expect to see `cast<T>(op)` used instead of `dynamic_cast<T>`